### PR TITLE
add binary py file and improved the song

### PR DESCRIPTION
**Pull request by xianwinwin**
the song first verse if up and running correctly. 
Plus, added py file.

### CODE REVIEW

File: binary_search.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import random
+
+
+class BinarySearch:
+
+	def __init__(self, arry):
+		self.arry = arry
+
+	def find(self, n):
+		l,r = 0, len(self.arry)-1
```
Comment (adhg) [inline binary_search.py:10]: yes this is correct, you should have it like this

File: binary_search.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		counter = 0
+
+		while l<r:
+			counter += 1
+			m = l + ((r-l)//2)
```
Comment (adhg) [inline binary_search.py:15]: excellent  - will also work with C++ and Java

File: binary_search.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+				print ("FOUND IT, counter:=",counter)
+				return True
+			if arry[m]>n:
+				r = m - 1
+			elif arry[m]<n:
+				l = m + 1			
+ 
+		print ("DID NOT FIND IT, countery:=",counter)
+		return False
+
```
Comment (adhg) [inline binary_search.py:26]: I think we can do better

File: binary_search.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+if __name__=='__main__':
+	print ("Start...")
+
+	arry_len = 300
+	arry = [random.randint(0, arry_len*2) for _ in range(0,arry_len)]
+	arry.sort() #nlogn	
+	bs = BinarySearch(arry)
+	f = bs.find(80)
+	print ('found it',f)
```
Comment (adhg) [inline binary_search.py:36]: find a number that is in the arry so we have a True

**Review by adhg — CHANGES_REQUESTED** [review]

File: binary_search.py (hunk 3 above)
Comment (xianwinwin) [reply binary_search.py:26]: meaning to return a dictionary?

**Comment by xianwinwin** [comment]
just <b> to go there </b> is bold message :-) +1

File: binary_search.py (hunk 4 above)
Comment (xianwinwin) [reply binary_search.py:36]: ok changed to 79; hopefully its there!

**Review by adhg — APPROVED** [review]

**Pull request merged**